ocs_ci/ocs/bucket_utils.py

```diff
@@ -158,6 +158,45 @@ def retrieve_anon_s3_resource():
     return anon_s3_resource
 
 
+def list_objects_from_bucket(
+    podobj, target, prefix=None, s3_obj=None, signed_request_creds=None, **kwargs
```

Comment:
Why not use s3_list_object_v1 or v2?, they are lightweight

Reply:
they dont really work like how the s3 ls works as mentioned in BZ.

Reply:
In what way it doesn't work as mentioned in bz? Nothing wrong with this implementation but the operation would be much quicker.

Reply:
This is the kind of output I get when I use s3_list_objects_v2():

> {'ResponseMetadata': {'RequestId': 'l3b8o9h5-e29xei-dn7', 'HostId': 'l3b8o9h5-e29xei-dn7', 'HTTPStatusCode': 200, 'HTTPHeaders': {'x-amz-request-id': 'l3b8o9h5-e29xei-dn7', 'x-amz-id-2': 'l3b8o9h5-e29xei-dn7', 'access-control-allow-origin': '*', 'access-control-allow-credentials': 'true', 'access-control-allow-methods': 'GET,POST,PUT,DELETE,OPTIONS', 'access-control-allow-headers': 'Content-Type,Content-MD5,Authorization,X-Amz-User-Agent,X-Amz-Date,ETag,X-Amz-Content-Sha256', 'access-control-expose-headers': 'ETag,X-Amz-Version-Id', 'content-type': 'application/xml', 'content-length': '925', 'date': 'Wed, 18 May 2022 07:04:49 GMT', 'keep-alive': 'timeout=5', 'set-cookie': '1a4aa612fe797ac8466d7ee00e5520d5=bd9231258a7d868b78a28e6d6f2d69d8; path=/; HttpOnly', 'cache-control': 'private'}, 'RetryAttempts': 0}, 'IsTruncated': False, 'Contents': [{'Key': 'mrbts/20220513/bbb/test-0', 'LastModified': datetime.datetime(2022, 5, 18, 7, 4, 49, tzinfo=tzutc()), 'ETag': '"e66b888473948b6644740e0b0e6da087"', 'Size': 1048576, 'StorageClass': 'STANDARD', 'Owner': {'DisplayName': 'NooBaa', 'ID': '123'}}, {'Key': 'mrbts/20220513/bb/test-0', 'LastModified': datetime.datetime(2022, 5, 18, 7, 4, 45, tzinfo=tzutc()), 'ETag': '"e66b888473948b6644740e0b0e6da087"', 'Size': 1048576, 'StorageClass': 'STANDARD', 'Owner': {'DisplayName': 'NooBaa', 'ID': '123'}}], 'Name': 's3-bucket-8a16d6e400f5493aa0dd609ebe5afc', 'Prefix': 'mrbts/20220513/', 'MaxKeys': 1000, 'KeyCount': 2, 'ContinuationToken': ''}

Reply:
But  `s3 ls` command gives different type of output and thats the one used in the BZ as well. soo :smile: @parikshithb

Reply:
ack